src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/PhysicalMemoryMonitor.Unix.cs

```diff
@@ -2,14 +2,53 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.IO;
+using System.Text.RegularExpressions;
 
 namespace System.Runtime.Caching
 {
     internal sealed partial class PhysicalMemoryMonitor : MemoryMonitor
     {
+        /* There are sysconf and /proc/meminfo ways to get this information before .Net 5,
+         * but it is very complicated to do it correctly, especially when accounting for
+         * container scenarios. The GC does this for us in .Net 5.
+         *
+         * Note: This is still a little bit off in some restrited memory scenarios, as
+         * 'TotalAvailableMemoryBytes' does not exactly report all of the available bytes.
+         * But if running in a memory-restricted Linux container, this "good citizen"
+         * memory monitor feels a little redundant anyway since memory is already capped
+         * via the system.
+         * But this comment (https://github.com/dotnet/coreclr/pull/25437#discussion_r299810957)
+         * highlights how our behavior might be slightly different from windows in these
+         * cases since this was a monitor that cared about actual physical memory.
+         */
+
+#if NET5_0
+        private long lastGCIndex;
+
         protected override int GetCurrentPressure()
         {
-            return 0;
+            // Get stats from GC. Try to get them to refresh if they haven't been updated since our last check.
+            GCMemoryInfo memInfo = GC.GetGCMemoryInfo();
+            if (memInfo.Index <= lastGCIndex)
+            {
+                GC.Collect(0); // A quick, ephemeral Gen 0 collection
```

Comment:
Consider using the overload which takes a GCCollectionMode and pass GCCollectionMode.Optimized. This allows the GC to determine when only a trivial amount of allocation has happened since the last GC and to skip it. In other words, if the memory situation hasn't have changed an appreciable amount, it skips the GC.